File: tugidigi-annotation/web/src/domain/targetimage.ts
export interface TargetImage{
version?:number;
id?:string;
binder?:string;
name?:string;
imageType?:string;
tags?:string[];
size?:ImageSize;
imagePath?:string;
note?:string;
bwThreshold?:number;
status?:Status;
annotations?:AnnotationObject[];
}
export interface ImageSize{
width?:number;
height?:number;
depth?:number;
}
export interface Bndbox{
xmin?:number;
ymin?:number;
xmax?:number;
ymax?:number;
}
export interface AnnotationObject{
id?:string;
name?:string;
bndbox?:Bndbox;
}
export type Status = "ESTIMATED"|"WORKING"|"CREATED"|"ANNOTATED"|"CHECKED";

File: tugidigi-annotation/src/main/java/jp/go/ndl/lab/annotation/domain/TargetImage.java
package jp.go.ndl.lab.annotation.domain;

import com.fasterxml.jackson.annotation.JsonInclude;
import jp.go.ndl.lab.annotation.infra.ESData;
import lombok.Data;

import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.ArrayList;
import java.util.List;

@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class TargetImage implements ESData {

    public Long version;
    public String id;
    public String binder;
    public String name;
    public String imageType;
    public List<String> tags;
    public ImageSize size = new ImageSize();
    public String annotator;

    public String imagePath;

    public Path toPath(){
        return Paths.get(imagePath);
    }

    public String note;

    public double bwThreshold = 0;

    public Status status;

    public List<AnnotationObject> annotations = new ArrayList<>();

    public void addTag(String tag) {
        if (tags == null) tags = new ArrayList<>();
        if (!tags.contains(tag)) tags.add(tag);
    }

}
